app/controllers/people_skills_controller.rb

```diff
@@ -23,18 +23,22 @@ def fetch_entries
                                   :category,
                                   :people, people_skills: :person
                                 ])
-    people_skills = PeopleSkillsFilter.new(base, params[:rated], params[:level]).scope
-    filter_entries(people_skills)
-  end
-
-  def filter_entries(people_skills)
     if params.key?(:person_id)
-      people_skills.where(person_id: params[:person_id])
+      fetch_person_entries(base)
     elsif params.key?(:skill_id)
-      people_skills.where(skill_id: params[:skill_id])
+      fetch_skill_entries(base)
     end
   end
 
+  def fetch_person_entries(base)
+    people_skills = PeopleSkillsFilter.new(base, params[:rated]).scope
+    people_skills.where(person_id: params[:person_id])
+  end
+
+  def fetch_skill_entries(base)
+    PeopleSkillsFilter.new(base, params[:rated], params[:level], params[:skill_id]).scope
+  end
```

Comment:
Just call the `PeopleSkillsFilter.new(...).scope` once like on line 39 and then if there is a :person_id param, just do the where on the result